### ISSUE TITLE
Missing istio_tcp_received_bytes_total in a sidecars namespace in Ambient profile

### ISSUE THREAD

**Issue by josunect** [istio/istio#50076]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

There is no telemetry data for `istio_tcp_received_bytes_total` for Telnet traffic in a namespace with sidecars in an Ambient profile, in Istio Ambient: 

Prometheus metrics: 

![image](https://github.com/istio/istio/assets/49480155/eed9476a-d36a-4006-90b9-23d38a3f84a2)

This can also be seen in the Kiali graph: 

![image](https://github.com/istio/istio/assets/49480155/710a011d-2a30-437b-90e2-7c86e5e8ed0b)

It also happens with istio_tcp_sent_bytes_total

![image](https://github.com/istio/istio/assets/49480155/f4d37d56-f9ab-4fb2-9b41-f9a1d797bdc8)

With Istio 1.21 and default profile, it is possible to see the `tcp_received_bytes_total`: 

![image](https://github.com/istio/istio/assets/49480155/7a75279f-318f-49b1-8126-6e0b17f878a8)

With this metrics it is possible also to see the traffic with Kiali: 
![image](https://github.com/istio/istio/assets/49480155/2f646d9c-98db-48a4-95c9-3eea29b84581)

### Version

```Text
$ istioctl version

client version: 1.21.0
control plane version: 1.21.0
data plane version: 1.21.0 (8 proxies)

$ kubectl version

Client Version: version.Info{Major:"1", Minor:"27"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"27"}
```

### Additional Information

How to reproduce it: 

(Tested in kind cluster with istio 1.21 and ambient profile)

```
kubectl create ns ambient
kubectl create ns sidecars

# Label namespace
kubectl label namespace sidecars istio-injection=enabled
kubectl label namespace ambient istio.io/dataplane-mode=ambient

# Create apps
kubectl apply -f ~/dev/deployments/traffic.yaml -n ambient
kubectl apply -f ~/dev/deployments/traffic.yaml -n sidecars

# Install telnet
kubectl exec $(kubectl get pod -l app=debugbox -n ambient -o jsonpath={.items..metadata.name}) -c debugbox -n ambient -- apt install telnet -y
kubectl exec $(kubectl get pod -l app=debugbox -n sidecars -o jsonpath={.items..metadata.name}) -c debugbox -n sidecars -- apt install telnet -y

# http traffic
kubectl exec $(kubectl get pod -l app=debugbox -n ambient -o jsonpath={.items..metadata.name}) -c debugbox -n ambient -- curl http://httpserver:8000/ip -s -o /dev/null -w "debugbox.test to httpserver: %{http_code}\n"

kubectl exec $(kubectl get pod -l app=debugbox -n sidecars -o jsonpath={.items..metadata.name}) -c debugbox -n sidecars -- curl http://httpserver:8000/ip -s -o /dev/null -w "debugbox.test to httpserver: %{http_code}\n"

# tcp traffic
kubectl exec -it $(kubectl get pod -l app=debugbox -n ambient -o jsonpath={.items..metadata.name}) -c debugbox -n ambient -- telnet tcpserver 9000
kubectl exec -it $(kubectl get pod -l app=debugbox -n sidecars -o jsonpath={.items..metadata.name}) -c debugbox -n sidecars -- telnet tcpserver 9000
```

Traffic.yaml: 
```
apiVersion: v1
kind: ServiceAccount
metadata:
  name: httpserver
---
apiVersion: v1
kind: Service
metadata:
  name: httpserver
  labels:
    app: httpserver
spec:
  ports:
  - name: http
    port: 8000
    targetPort: 80
  selector:
    app: httpserver
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: httpserver
spec:
  replicas: 1
  selector:
    matchLabels:
      app: httpserver
      version: v1
  template:
    metadata:
      labels:
        app: httpserver
        version: v1
    spec:
      containers:
      - image: docker.io/kennethreitz/httpbin
        imagePullPolicy: IfNotPresent
        name: httpserver
        ports:
        - containerPort: 80
---
apiVersion: v1
kind: ServiceAccount
metadata:
  name: debugbox
---
apiVersion: v1
kind: Service
metadata:
  name: debugbox
  labels:
    app: debugbox
spec:
  ports:
  - port: 80
    name: http
  selector:
    app: debugbox
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: debugbox
spec:
  replicas: 1
  selector:
    matchLabels:
      app: debugbox
  template:
    metadata:
      labels:
        app: debugbox
    spec:
      containers:
      - name: debugbox
        image: markeijsermans/debug
        command: ["/bin/sleep", "3650d"]
        imagePullPolicy: IfNotPresent
---
apiVersion: v1
kind: Service
metadata:
  name: tcpserver
  labels:
    app: tcpserver
spec:
  ports:
  - port: 9000
    name: tcp
  selector:
    app: tcpserver
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: tcpserver
spec:
  replicas: 1
  selector:
    matchLabels:
      app: tcpserver
  template:
    metadata:
      labels:
        app: tcpserver
    spec:
      containers:
      - name: tcpserver
        image: istio/tcp-echo-server:1.1
        imagePullPolicy: IfNotPresent
        args: [ "9000", "hello" ]
        ports:
        - containerPort: 9000
---
apiVersion: v1
kind: ServiceAccount
metadata:
  name: tcpserver
```

**Comment by josunect** [comment]
Without deploying anything in the Ambient namespace there is no ztunnel metrics: 

```
# HELP istio_build Istio component build info.
# TYPE istio_build gauge
istio_build{component="ztunnel",tag="unknown"} 1
# HELP istio_connection_terminations The total number of completed connections to xds server.
# TYPE istio_connection_terminations counter
# HELP istio_tcp_connections_opened The total number of TCP connections opened.
# TYPE istio_tcp_connections_opened counter
# HELP istio_tcp_connections_closed The total number of TCP connections closed.
# TYPE istio_tcp_connections_closed counter
# HELP istio_tcp_received_bytes The size of total bytes received during request in case of a TCP connection.
# TYPE istio_tcp_received_bytes counter
# HELP istio_tcp_sent_bytes The size of total bytes sent during response in case of a TCP connection.
# TYPE istio_tcp_sent_bytes counter
# HELP istio_on_demand_dns The total number of requests that used on-demand DNS.
# TYPE istio_on_demand_dns counter
# HELP istio_on_demand_dns_cache_misses The total number of cache misses for requests on-demand DNS.
# TYPE istio_on_demand_dns_cache_misses counter
# HELP workload_manager_active_proxy_count The total number current workloads with active proxies.
# TYPE workload_manager_active_proxy_count gauge
workload_manager_active_proxy_count{} 0
# HELP workload_manager_pending_proxy_count The total number current workloads with pending proxies.
# TYPE workload_manager_pending_proxy_count gauge
workload_manager_pending_proxy_count{} 0
# HELP workload_manager_proxies_started The total number of proxies that were started.
# TYPE workload_manager_proxies_started counter
# HELP workload_manager_proxies_stopped The total number of proxies that were stopped.
# TYPE workload_manager_proxies_stopped counter
# EOF
```

After deployment the Ambient namespace there are metrics (You can see the metrics corresponding to the http requests): 

```

# HELP istio_build Istio component build info.
# TYPE istio_build gauge
istio_build{component="ztunnel",tag="unknown"} 1
# HELP istio_connection_terminations The total number of completed connections to xds server.
# TYPE istio_connection_terminations counter
# HELP istio_tcp_connections_opened The total number of TCP connections opened.
# TYPE istio_tcp_connections_opened counter
istio_tcp_connections_opened_total{reporter="destination",source_workload="debugbox",source_canonical_service="debugbox",source_canonical_revision="latest",source_workload_namespace="ambient",source_principal="spiffe://cluster.local/ns/ambient/sa/default",source_app="debugbox",source_version="latest",source_cluster="Kubernetes",destination_service="tcpserver.ambient.svc.cluster.local",destination_service_namespace="ambient",destination_service_name="tcpserver",destination_workload="tcpserver",destination_canonical_service="tcpserver",destination_canonical_revision="latest",destination_workload_namespace="ambient",destination_principal="spiffe://cluster.local/ns/ambient/sa/default",destination_app="tcpserver",destination_version="latest",destination_cluster="Kubernetes",request_protocol="tcp",response_flags="-",connection_security_policy="mutual_tls"} 1
istio_tcp_connections_opened_total{reporter="destination",source_workload="debugbox",source_canonical_service="debugbox",source_canonical_revision="latest",source_workload_namespace="ambient",source_principal="spiffe://cluster.local/ns/ambient/sa/default",source_app="debugbox",source_version="latest",source_cluster="Kubernetes",destination_service="httpserver.ambient.svc.cluster.local",destination_service_namespace="ambient",destination_service_name="httpserver",destination_workload="httpserver",destination_canonical_service="httpserver",destination_canonical_revision="v1",destination_workload_namespace="ambient",destination_principal="spiffe://cluster.local/ns/ambient/sa/default",destination_app="httpserver",destination_version="v1",destination_cluster="Kubernetes",request_protocol="tcp",response_flags="-",connection_security_policy="mutual_tls"} 5
# HELP istio_tcp_connections_closed The total number of TCP connections closed.
# TYPE istio_tcp_connections_closed counter
istio_tcp_connections_closed_total{reporter="destination",source_workload="debugbox",source_canonical_service="debugbox",source_canonical_revision="latest",source_workload_namespace="ambient",source_principal="spiffe://cluster.local/ns/ambient/sa/default",source_app="debugbox",source_version="latest",source_cluster="Kubernetes",destination_service="httpserver.ambient.svc.cluster.local",destination_service_namespace="ambient",destination_service_name="httpserver",destination_workload="httpserver",destination_canonical_service="httpserver",destination_canonical_revision="v1",destination_workload_namespace="ambient",destination_principal="spiffe://cluster.local/ns/ambient/sa/default",destination_app="httpserver",destination_version="v1",destination_cluster="Kubernetes",request_protocol="tcp",response_flags="-",connection_security_policy="mutual_tls"} 5
# HELP istio_tcp_received_bytes The size of total bytes received during request in case of a TCP connection.
# TYPE istio_tcp_received_bytes counter
istio_tcp_received_bytes_total{reporter="destination",source_workload="debugbox",source_canonical_service="debugbox",source_canonical_revision="latest",source_workload_namespace="ambient",source_principal="spiffe://cluster.local/ns/ambient/sa/default",source_app="debugbox",source_version="latest",source_cluster="Kubernetes",destination_service="httpserver.ambient.svc.cluster.local",destination_service_namespace="ambient",destination_service_name="httpserver",destination_workload="httpserver",destination_canonical_service="httpserver",destination_canonical_revision="v1",destination_workload_namespace="ambient",destination_principal="spiffe://cluster.local/ns/ambient/sa/default",destination_app="httpserver",destination_version="v1",destination_cluster="Kubernetes",request_protocol="tcp",response_flags="-",connection_security_policy="mutual_tls"} 405
# HELP istio_tcp_sent_bytes The size of total bytes sent during response in case of a TCP connection.
# TYPE istio_tcp_sent_bytes counter
istio_tcp_sent_bytes_total{reporter="destination",source_workload="debugbox",source_canonical_service="debugbox",source_canonical_revision="latest",source_workload_namespace="ambient",source_principal="spiffe://cluster.local/ns/ambient/sa/default",source_app="debugbox",source_version="latest",source_cluster="Kubernetes",destination_service="httpserver.ambient.svc.cluster.local",destination_service_namespace="ambient",destination_service_name="httpserver",destination_workload="httpserver",destination_canonical_service="httpserver",destination_canonical_revision="v1",destination_workload_namespace="ambient",destination_principal="spiffe://cluster.local/ns/ambient/sa/default",destination_app="httpserver",destination_version="v1",destination_cluster="Kubernetes",request_protocol="tcp",response_flags="-",connection_security_policy="mutual_tls"} 1290
# HELP istio_on_demand_dns The total number of requests that used on-demand DNS.
# TYPE istio_on_demand_dns counter
# HELP istio_on_demand_dns_cache_misses The total number of cache misses for requests on-demand DNS.
# TYPE istio_on_demand_dns_cache_misses counter
# HELP workload_manager_active_proxy_count The total number current workloads with active proxies.
# TYPE workload_manager_active_proxy_count gauge
workload_manager_active_proxy_count{} 2
# HELP workload_manager_pending_proxy_count The total number current workloads with pending proxies.
# TYPE workload_manager_pending_proxy_count gauge
workload_manager_pending_proxy_count{} 0
# HELP workload_manager_proxies_started The total number of proxies that were started.
# TYPE workload_manager_proxies_started counter
workload_manager_proxies_started_total{} 4
# HELP workload_manager_proxies_stopped The total number of proxies that were stopped.
# TYPE workload_manager_proxies_stopped counter
workload_manager_proxies_stopped_total{} 2
# EOF
```

All the scrape pools are "UP": 

![image](https://github.com/istio/istio/assets/49480155/3b424479-a4df-49e6-915f-44a26cc6992b)

**Comment by howardjohn** [comment]
Here is what I am getting on latest:

![2024-06-13_12-15-51](https://github.com/istio/istio/assets/623453/eeed2078-0453-43e8-aa43-0754784e7f92)

Which looks correct to me. There were a few telemetry fixes recently, might have been fixed in one of them?

**Comment by josunect** [comment]
It looks like it has been fixed (Tested with Istio 1.22.2)

Thanks!

![image](https://github.com/istio/istio/assets/49480155/7beb2869-27f9-4c57-891f-4986b0b0cd20)

**Comment by howardjohn** [comment]
great to hear, thanks for verifying!
